#! /bin/sh -e
## 66_donot_let_cache_html_pages.dpatch
## by László 'GCS' Böszörményi <gcs@lsc.hu>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Add 'Pragma: no-cache' to HTML page outputs, so proxies like Squid
## DP: do not cache them, and changes become visible immediately.

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch ${2:+-d $2}}"

if [ $# -lt 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch) patch $patch_opts -p1 < $0;;
    -unpatch) patch $patch_opts -p1 -R < $0;;
    *)
        echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
        exit 1;;
esac

exit 0
@DPATCH@
--- mailman-2.1.5/Mailman/htmlformat.py.orig	2003-09-22 02:58:13.000000000 +0000
+++ mailman-2.1.5/Mailman/htmlformat.py		2004-11-16 16:47:01.000000000 +0000
@@ -299,7 +299,8 @@
         charset = 'us-ascii'
         if self.language:
             charset = Utils.GetCharSet(self.language)
-        output = ['Content-Type: text/html; charset=%s\n' % charset]
+        output = ['Content-Type: text/html; charset=%s' % charset]
+        output.append('Cache-control: no-cache\n')
         if not self.suppress_head:
             kws.setdefault('bgcolor', self.bgcolor)
             tab = ' ' * indent
